app/models/order_cycle.rb

```diff
@@ -47,7 +47,7 @@ class OrderCycle < ActiveRecord::Base
     if user.has_spree_role?('admin')
       scoped
     else
-      where('coordinator_id IN (?)', user.enterprises)
+      where('coordinator_id IN (?)', user.enterprises.map(&:id))
```

Comment:
Have we checked we are not affecting performance here? When passing the association I guess AR calls `enterprises_ids` which will do clever things, whereas with map we load the entire association and then iterate it.

I'd like to understand what's the actual change we are doing.

Reply:
the error we are fixing is that we can't have enterprises matching coordinator_id, we need enterprises ids. it just fails in rails 4.
can we use user.enterprises_ids? instead of user.enterprises.map(&:id)?
we could use select or map, but this case is better with map because the user.enterprises is already lazy loaded, with select we would trigger another db query.